docs: add eslint setup guide (#24976)
nuxt/nuxt: docs/2.guide/1.concepts/9.code-style.md

---
title: 'Code Style'

> description: "Nuxt 3 supports eslint out of the box"

Reviewer: ```suggestion
description: "Nuxt supports ESLint out of the box"
```

---

> ## Eslint

Reviewer: ```suggestion
## ESLint
```

> The recommended approach for Nuxt is to enable eslint using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config). 

Reviewer: ```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config). 
```

**Install**

  ```bash [yarn]
  yarn add --dev eslint @antfu/eslint-config
  ```

  ```bash [npm]
  npm install --save-dev eslint @antfu/eslint-config
  ```

  ```bash [pnpm]
  pnpm add -D eslint @antfu/eslint-config
  ```

  ```bash [bun]
  bun add -D eslint @antfu/eslint-config
  ```

> **Add eslint.config.js to the root folder of your Nuxt app**

Reviewer 2: ```suggestion
**Add `eslint.config.js` to the root folder of your Nuxt app**
```

```js
module.exports = {
  root: true,
  extends: ['@nuxt/eslint-config'],
}
```

## Modify package.json

Add the below to lint commands to your `package.json` script section:

```json
  "scripts": {
    ...
    "lint": "eslint .",
    "lint:fix": "eslint . --fix",
    ...
  },
```

> Run the lint command to check if the codestyle is correct or run `lint:fix` to automatically fix issues.

Reviewer: ```suggestion
Run the lint command to check if the code style is correct or run `lint:fix` to automatically fix issues.
```

## Configuring VSCode

Install the [VSCode ESLint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint).

In VSCode press `ctrl+shift+p` (`cmd+shift+p` on Mac) to open the command prompt, find `Open Workspace Settings (JSON)`, add the below lines to the JSON and save:

```json
{

>   "editor.formatOnSave": true,

Reviewer 3: I am not sure if enabling this has something to do with ESLint (`codeActionsOnSave` is different from `formatOnSave`, I believe). In my own usage, I set this off to disable Prettier in favor of ESLint's auto-fix. I guess for `@nuxt/eslint-config`, we don't need to mention this line as there is no formatting involved.

```suggestion
```

  "editor.codeActionsOnSave": {

>     "source.fixAll.eslint": true

Reviewer 3: ```suggestion
    "source.fixAll.eslint": "explicit"
```

It's updated in the latest VS Code

  }
}
```

> You're good to go! On save, your files will be formatted. 

Reviewer 3: ```suggestion
You're good to go! On file save, your files will be lint with auto-fix.
```

We should maybe mention that `@nuxt/eslint-config` does not do auto-formatting for you, and maybe guide ppl to set up Prettier?

---
title: 'Code Style'
description: "Nuxt supports ESLint out of the box"
---

## ESLint

> The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).

Reviewer: what about this @antfu?

```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).

At the moment, this configuration will not format your files; you can set up Prettier or another tool to do so.

::alert{type=info}
We're currently working to refactor the Nuxt ESLint configuration. Subscribe to [the Nuxt ESLint roadmap](https://github.com/nuxt/eslint-config/issues/303) to follow updates.]
::
```

---
title: 'Code Style'
description: "Nuxt supports ESLint out of the box"
---

## ESLint

> The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config). 

Reviewer: what about this @antfu?

```suggestion
The recommended approach for Nuxt is to enable ESLint support using [`@nuxt/eslint-config`](https://github.com/nuxt/eslint-config).

At the moment, this configuration will not format your files; you can set up Prettier or another tool to do so.

::alert{type=info}
We're currently working to refactor the Nuxt ESLint configuration. Subscribe to [the Nuxt ESLint roadmap](https://github.com/nuxt/eslint-config/issues/303) to follow updates.]
::
```

### Install Dependencies

Install both ESLint and the Nuxt configuration as development dependencies.

::code-group

```bash [yarn]
yarn add --dev eslint @nuxt/eslint-config
```

```bash [npm]
npm install --save-dev eslint @nuxt/eslint-config
```

```bash [pnpm]
pnpm add -D eslint @nuxt/eslint-config
```

```bash [bun]
bun add -D eslint @nuxt/eslint-config
```

::

### Configuration

Add `.eslintrc.cjs` to the root folder of your Nuxt app.

```js
module.exports = {
  root: true,
  extends: ['@nuxt/eslint-config'],
}
```

### Modify package.json

Add the below to lint commands to your `package.json` script section:

```json
  "scripts": {
    ...
    "lint": "eslint .",
    "lint:fix": "eslint . --fix",
    ...
  },
```

> Run the lint command to check if the code style is correct or run `lint:fix` to automatically fix issues.

Reviewer: ```suggestion
Run the `lint` command to check if the code style is correct or run `lint:fix` to automatically fix issues.
```
